### ISSUE TITLE
Copy note's anchor/title as universal callback URL

### ISSUE THREAD

**Issue by executed** [benlau/joplin-plugin-ianylink#2]
Your project inspired me to make Joplin support your plugin even further.

Right now Joplin supports following callback URLs:
https://joplinapp.org/help/apps/external_links/

What I want to introduce in Joplin core code is external link targeting note's anchor/title, which is especially useful in case of long notes.
`joplin://x-callback-url/openItem?id=ad768a6768564d2bbb2cb87d88b82b31&hash=some-title`

It seems to be working locally, still need to figure out how to make it work when note is already opened and when app is not running yet.

Now if it makes sense I wonder how to let users easily decide whether they want to copy link to anchor as Markdown Link, or as Universal Link.

Maybe good solution would be to add 2 options into right-click context menu of both MD editor and viewer when right click happens on element that looks like H1/H2/H3/etc.
Similarly 2 keyboard shortcuts could be added so that when triggered from the line that looks like H1/H2/H3/etc the Markdown Link/Universal Link is copied to the clipboard.

Ben, what do you think about this?

**Comment by executed** [comment]
I should also mention that pre-requisites are not only working anchor callback URLs in Joplin but also fixed anchor navigation itself which is very bad currently in the latest version.
I think I have a fix for it too, but I lack the experience and hardware to test it well on different platforms.

**Comment by benlau** [comment]
Do you mean something like this plugin?

https://discourse.joplinapp.org/t/plugin-copy-anchor-link/21690

**Comment by executed** [comment]
@benlau 
Yes, but instead of one copy icon - I think it should reveal 2 options on right-click - Copy Markdown Link & Copy Universal Link

Additionally, same 2 options should be available as keyboard shortcuts

**Comment by executed** [comment]
Ah, I see. It gives you multiple icons... Yes, it would make sense to adjust that plugin to provide universal web link as one of icons. 

To summarize, here are things needed for copying note's anchor/title as universal callback URLs:
- [x] Fix Joplin Core: sudden scrolling to top of note when using markdown link

- [ ] Enhance Joplin Core: add support of note's anchor/title navigation through callback URLs

- [ ] Enhance Copy Anchor Link plugin: add universal link support to plugin's anchor icons

Optional:
- [ ] Enhance Copy Anchor Link plugin: define keyboard shortcuts for each type of link, including a universal link
